dsp/transform/hilbert.go

```diff
@@ -0,0 +1,76 @@
+// Copyright ©2024 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package transform
+
+import (
+	"gonum.org/v1/gonum/dsp/fourier"
+)
+
+// Hilbert implements an approximate Hilbert transform that allows calculation
+// of an approximate analytical signal of a real signal, and determine the
+// real envelope of a signal.
+type Hilbert struct {
+	fft  *fourier.CmplxFFT
+	work []complex128
+}
+
+// NewHilbert returns a new Hilbert transformer for signals of size n.
+// The transform is most efficient when n is a product of small primes.
+// If n is 0 or negative, the function will panic.
```

Comment:
```suggestion
// n must not be less than one.
```
I would prefer to state what a user cannot do than what we guarantee to do.

(though I don't think we make this statement in the cause for the panic; I need to think about how I actually want to handle that)